File: packages/forge-core/src/forge_core/domain/entities/battlefield.py
from .base import CamelModel as BaseModel

from .enums import CardClass, TerrainType


class GridPosition(BaseModel):
    row: int
    col: int


class TerrainModifier(BaseModel):
    target_class: CardClass
    multiplier: float


class BattlefieldSlot(BaseModel):
    slot_id: str
    position: GridPosition
    terrain_type: TerrainType | None = None
    modifiers: list[TerrainModifier] = []


class BattlefieldGrid(BaseModel):
    rows: int
    cols: int


class BattlefieldConfig(BaseModel):
    id: str
    format_version: str
    name: str
    grid: BattlefieldGrid
    slots: list[BattlefieldSlot]

File: packages/ts/shared-schema/src/entities/battlefield.ts
import { TerrainType } from '../enums/terrain-type';
import { CardClass } from '../enums/card-class';

export interface GridPosition {
  row: number;
  col: number;
}

export interface TerrainModifier {
  targetClass: CardClass;
  multiplier: number;
}

export interface BattlefieldSlot {
  slotId: string;
  position: GridPosition;
  terrainType: TerrainType | null;
  modifiers: TerrainModifier[];
}

export interface BattlefieldGrid {
  rows: number;
  cols: number;
}

export interface BattlefieldConfig {
  id: string;
  formatVersion: string;
  name: string;
  grid: BattlefieldGrid;
  slots: BattlefieldSlot[];
}
